Fixing DB schema error

josenavas opened the pull request:
@AmandaBirmingham brought to my attention that the gDNA extraction process and amplicon library processes where used incorrectly. I introduced a set of tables in January 17 that broke the actual assumption behind those processes, so here I fix the DB schema, fix the objects and the interface to behave as expected.

In short: each process instance applies to a single plate (rather than a set of plates as I had it before).

@AmandaBirmingham this is the DB structure define here, does it correctly match your expectations?

![screen shot 2018-01-24 at 2 12 57 pm](https://user-images.githubusercontent.com/2501478/35359941-ca92a052-0110-11e8-8a9f-c41a5509f248.png)

josenavas commented:
I forgot to mention that the actual interface behavior has not changed.

josenavas commented:
@ElDeveloper Can you take a quick look in here? This is blocking some other development and I would like to start moving forward.

ElDeveloper commented:
On it.

josenavas commented:
Awesome thanks!

AmandaBirmingham commented:
Yes, that matches my expectations.  Strictly speaking, each process applies to one or more wells—but in practice so far that generally equates to one process per plate.

From: Jose Navas <notifications@github.com>
Reply-To: jdereus/labman <reply@reply.github.com>
Date: Wednesday, January 24, 2018 at 2:15 PM
To: jdereus/labman <labman@noreply.github.com>
Cc: "Birmingham, Amanda" <abirmingham@ucsd.edu>, Mention <mention@noreply.github.com>
Subject: [jdereus/labman] Fixing DB schema error (#124)


@AmandaBirmingham<https://github.com/amandabirmingham> brought to my attention that the gDNA extraction process and amplicon library processes where used incorrectly. I introduced a set of tables in January 17 that broke the actual assumption behind those processes, so here I fix the DB schema, fix the objects and the interface to behave as expected.

In short: each process instance applies to a single plate (rather than a set of plates as I had it before).

@AmandaBirmingham<https://github.com/amandabirmingham> this is the DB structure define here, does it correctly match your expectations?

[screen shot 2018-01-24 at 2 12 57 pm]<https://user-images.githubusercontent.com/2501478/35359941-ca92a052-0110-11e8-8a9f-c41a5509f248.png>

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/jdereus/labman/pull/124

Commit Summary

  *   Fixing DB structure
  *   Fixing the gdna extraction process
  *   Fixing 16S process
  *   Adding sample plate attribute to the gdna extraction process
  *   Adding volume to gdna extraction
  *   Fixing gdna extraction interface
  *   Adding missing properties to 16S lib process
  *   Fixing interface

File Changes

  *   M labman/db/process.py<https://github.com/jdereus/labman/pull/124/files#diff-0> (436)
  *   M labman/db/support_files/db_patch.sql<https://github.com/jdereus/labman/pull/124/files#diff-1> (88)
  *   M labman/db/support_files/labman.dbs<https://github.com/jdereus/labman/pull/124/files#diff-2> (110)
  *   M labman/db/support_files/labman.html<https://github.com/jdereus/labman/pull/124/files#diff-3> (1004)
  *   M labman/db/support_files/populate_test_db.sql<https://github.com/jdereus/labman/pull/124/files#diff-4> (14)
  *   M labman/db/tests/test_process.py<https://github.com/jdereus/labman/pull/124/files#diff-5> (70)
  *   M labman/gui/handlers/process_handlers/gdna_extraction_process.py<https://github.com/jdereus/labman/pull/124/files#diff-6> (47)
  *   M labman/gui/handlers/process_handlers/library_prep_16s_process.py<https://github.com/jdereus/labman/pull/124/files#diff-7> (59)
  *   M labman/gui/handlers/process_handlers/test/test_gdna_extraction_process.py<https://github.com/jdereus/labman/pull/124/files#diff-8> (9)
  *   M labman/gui/handlers/process_handlers/test/test_library_prep_16s_process.py<https://github.com/jdereus/labman/pull/124/files#diff-9> (6)
  *   M labman/gui/templates/extraction.html<https://github.com/jdereus/labman/pull/124/files#diff-10> (42)
  *   M labman/gui/templates/library_prep_16S.html<https://github.com/jdereus/labman/pull/124/files#diff-11> (38)

Patch Links:

  *   https://github.com/jdereus/labman/pull/124.patch
  *   https://github.com/jdereus/labman/pull/124.diff

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/jdereus/labman/pull/124>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AKLur1MzxtOsJ7t6goYBUhSfOlA7jsOIks5tN6s3gaJpZM4RsBBP>.

josenavas commented:
Thanks for confirming @AmandaBirmingham !

ElDeveloper commented on labman/gui/templates/library_prep_16S.html:
```diff
@@ -9,27 +9,35 @@
-        });
-        disableAll();
+        // Need to add a timeout so the interface is ready to set the values
+        setTimeout(function() {
```
Is a timeout needed here or would a better way be to wait using:

```javascript
$(function() {
  // things that update the GUI
});
```

Or is the timeout waiting for something else?

josenavas replied:
I don't think that would work. The interface that I'm waiting for here is created in the previous "addPlate" call, which executes an AJAX query, and builds the interface based on the results of the ajax query. This function already gets executed _after_ the ajax called is completed, but it is not ensured that the html that the function create is ready yet.

ElDeveloper replied:
I see, thanks for the info!

josenavas commented:
You're right @tanaes - everything else that was focusing in a single plate was working as expected.

Unfortunately there is a second DB modification that I'm working on right now, so I hope it will be fixed by the time everybody starts working in the morning (except probably @antgonza and @jdereus 😄 ).

In short, the compress gDNA plate information was getting lost (i.e. which plate they originated from). After discussing with @AmandaBirmingham we have decided that the best thing to do is to create a new composition and a new process type to store this information. This is analogous to the normalized gdna process/composition. I've already modified the DB, created the new composition object and fixed the composition tests. I'm working on creating the process object and I will be updating the rest of the code as needed.
